test/integration/consul_test.go

```diff
@@ -0,0 +1,106 @@
+//go:build integration
+// +build integration
+
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-consul/src/args"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	containerIntegration = "nri-consul"
+	schema               = "consul-schema.json"
+)
+
+var (
+	containersConsul = []string{"consul-server1", "consul-server2", "consul-server3"}
+	responseRegex    = regexp.MustCompile(`{"name":"com\.newrelic\.consul".*`)
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	result := m.Run()
+	os.Exit(result)
+}
+
+func TestMetricsCollectionInCluterWithSSL(t *testing.T) {
+	if !waitForConsulClusterUpAndRunning(20) {
```

Comment:
I think we use the [depends_on](https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on) feature of docker-compose to achieve similar things , have you tried that? here is an example https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on